src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/LockFileUtils.cs

```diff
@@ -229,6 +216,44 @@ internal static List<List<SelectionCriteria>> CreateOrderedCriteriaSets(RestoreT
             ApplyReferenceFilter(lockFileLib, framework, nuspec);
         }
 
+        private static void AddMSBuildAssets(
+            LockFileLibrary library,
+            RestoreTargetGraph targetGraph,
+            LockFileTargetLibrary lockFileLib,
+            IReadOnlyList<SelectionCriteria> orderedCriteria,
+            ContentItemCollection contentItems)
+        {
+            // Build Transitive
+            var btGroup = GetLockFileItems(
+                orderedCriteria,
+                contentItems,
+                targetGraph.Conventions.Patterns.MSBuildTransitiveFiles);
+
+            var filteredBTGroup = GetBuildItemsForPackageId(btGroup, library.Name);
+            lockFileLib.Build.AddRange(filteredBTGroup);
+
+            // Build
+            var buildGroup = GetLockFileItems(
+                orderedCriteria,
+                contentItems,
+                targetGraph.Conventions.Patterns.MSBuildFiles);
+
+            // filter any build asset already being added as part of build transitive
+            var filteredBuildGroup = GetBuildItemsForPackageId(buildGroup, library.Name).
+                Where(buildItem => !filteredBTGroup.Any(
+                    btItem => Path.GetFileName(btItem.Path).Equals(Path.GetFileName(buildItem.Path), StringComparison.OrdinalIgnoreCase)));
```

Comment:
because `/buildTransitive` assets will be duplicated in `/build` folder as well (for PC compatibility) and we don't want to add the same asset from both folders.

Reply:
There's no scenario where they'd be different?

Reply:
Suppose packageA has
`/buildTransitive/packageA.props`
`/build/packageA.props` and /build/packageA.targets`

Now, in the consuming project, it will be `/buildTransitive/packageA.props` and /build/packageA.targets`. We dont want `/build/packageA.props` as well hence this filtering.

Hope this clarifies!

Reply:
Yeah, makes sense. 
Basically there is no scenario where that's the wanted experience. 

I'm fine with the change as is.